src/thunderbots/software/ai/world/team.h

```diff
@@ -70,4 +120,38 @@ class Team
      * @return a vector of all the robots on this team.
      */
     std::vector<Robot> getAllRobots() const;
+
+    /**
+     * Removes all Robots from this team. Does not affect the goalie id.
+     */
+    void clearAllRobots();
+
+    /**
+     * Defines the equality operator for a Team. Teams are equal if their robots and
+     * goalies are equal
+     *
+     * @param other The team to compare against for equality
+     * @return True if the other team is equal to this team, and false otherwise
+     */
```

Comment:
Given that which robot is the goalie is just determined by the set id for the goalie (as I understand it), we should be able to simplify this to "Teams are equal if their robots are equal"?

Reply:
Teams with the same robots but with different goalies are not considered equal.

Team a = robot_0, robot_1, robot_2 and robot_0 is assigned to be the goalie
Team b = robot_0, robot_1, robot_2 and robot_1 is assigned to be the goalie
a != b

So we could say "Teams are equal if their robots are equal and have the same robot assigned as the goalie"